src/commands/functions/create.js

```diff
@@ -373,6 +367,42 @@ const createFunctionAddon = async function ({ api, addons, siteId, addonName, si
   }
 }
 
+const injectEnvVariables = async ({ context }) => {
+  const { log, warn, error, netlify } = context
+  const { api, site } = netlify
+  const { teamEnv, addonsEnv, siteEnv, dotFilesEnv } = await getSiteInformation({
+    api,
+    site,
+    warn,
+    error,
+  })
+  await addEnvVariables({ log, teamEnv, addonsEnv, siteEnv, dotFilesEnv })
+}
+
+const handleOnComplete = async ({ context, onComplete }) => {
+  if (onComplete) {
+    await injectEnvVariables({ context })
+    await onComplete.call(context)
```

Comment:
Passing the `context` as an argument instead of `this` might be simpler. Unfortunately, this would be a breaking change since the template file is supplied by the user, correct?

Reply:
Correct, see https://github.com/netlify/cli/blob/65c4bd70bc51492b49fbae677526aea881edbd93/src/functions-templates/js/url-shortener/.netlify-function-template.js#L17
We can update the templates in this repo, but this will still break templates installed from URLs.
At some point we can collect analytics on `functions:create` from URLs and see if this can break anything.